pennylane/ops/qubit/parametric_ops.py

```diff
@@ -312,9 +314,10 @@ def compute_eigvals(theta):  # pylint: disable=arguments-differ
         if qml.math.get_interface(theta) == "tensorflow":
             theta = qml.math.cast_like(theta, 1j)
 
-        p = qml.math.exp(-0.5j * theta)
+        if qml.math.ndim(theta) == 0:
+            return qml.math.exp(0.5j * theta * np.array([-1, 1]))
```

Comment:
These benchmarks were made with a single scalar phase (`np.pi`).

Reply:
Again, unfortunately this notation is not compatible with torch, because `qml.math.exp([-arg, arg])` will not return a `torch.tensor` object, but a `numpy.ndarray` of `torch.tensor` entries. In particular, the resulting function will not be differentiable by torch.